src/App.js

```diff
@@ -1,5 +1,44 @@
+import { OUTPUT_MESSAGE_WINNER } from './lib/constants.js';
+import { print } from './lib/utils.js';
+import Input from './Input.js';
+import Race from './Race.js';
+
 class App {
-  async run() {}
+  #carArray;
+
+  #tryCount;
+
+  #winnerCarArray;
+
+  constructor() {
+    this.#carArray = [];
+    this.#tryCount = [];
+    this.#winnerCarArray = [];
+  }
+
+  async run() {
+    await this.#getInput();
+    this.#runRace();
+    this.#printOutput();
+  }
+
+  async #getInput() {
+    const input = new Input();
+
+    await input.getUserInput();
+    this.#carArray = input.parseCars();
+    this.#tryCount = input.parseTryCount();
```

Comment:
`input` 두 개를 먼저 입력 받고, `parseCars`와 `parseTryCount`를 통해서 유효성검사를 하면 각각의 입력값에 대한 에러처리가 안되지 않나요? 이게 무슨 말이냐면... 사용자가 자동차 이름을 입력할 때 잘못입력하면, 시도횟수를 입력하고 나서야 에러가 발생할 것 같은데 맞나요? 

그리구 이 부분이 조금 코드상 이해하기 어려운 것 같습니다. `input.getUserInput()`을 해놓고 `parseCars`를 통해 `this.#carArray`에 담는게 어색한 것 같아요
```suggestion
    const carInput = await input.getUserInput();
    this.#carArray = input.parseCars(carInput);
    
    const tryInput = await input.getUserInput();
    this.#tryCount = input.parseTryCount(tryInput);
```
메서드 이름이 parse인데 파라미터에 전달되는게 없으니 뭐가 `parse`되는 건지 숨어있는 느낌이라고 해야 할까요..?

Reply:
상당히 치명적인 실수였던 거 같아요. 두 입력값을 받으면 그 때 유효성 체크를하는군요.

날카로운 지적 감사합니다.

두 번째 부분에서, 저는 Input이라는 객체에 초점을 맞췄어요. `Input`의 객체의 인스턴스인 `input`입장에서 보면 입력값을 받고, 이를 파싱하니 이상하지 않다고 생각해요.

하지만 지예님처럼 함수형 프로그래밍 관점에서 보면, 인자에 값을 넘기지 않고 파싱을 하는 것이 이상하다고 생각할 수 있을 거 같아요!

저도 이 부분에서 어떤게 정답인지는 아직 잘 모르겠지만, 위 부분에서는 `Input`이 클래스 즉 객체이고, 객체지향적스럽게 작성하기 위해 프로퍼티를 활용하려고 했던 거 같아요.

Reply:
저도 파싱하는 흐름이 살짝 이해되지 않아서 두분 코멘트를 읽어 보게 되었는데요..!
제 생각에는 입력을 받을 때에는 `carArray`와 `tryCount`를 `getUserInput`이라는 메소드 하나로 입력을 받았는데,
파싱 과정에서는 `parseCars`와 `parseTryCount`를 따로 호출하고 있어서 좀 헷갈리는 것 같습니다.
파싱 역시 하나의 메소드로 두개의 프로퍼티를 파싱할 수 있게 한다면 조금 덜 헷갈릴 수 있을 것 같아요.

```suggestion
    await input.getUserInput();
    const { carArray, tryCount } = input.parseUserInput();
    this.#carArray = carArray;
    this.#tryCount = tryCount;
```

Reply:
그렇군요. 유연님처럼 생각할 수도 있겠네요. 입력은 하나인데, 파싱은 두 번 처리하고 있군요. 그 부분까지 생각을 못했어요!

입력도 각각 받거나, 혹은 말씀해주신 것처럼 한 번에 파싱을 하는 방식이 좋을 거 같아요.

좋은 지적 감사합니다 ☺️

Reply:
유연님 의견이 정말 맞는 것 같습니다! 제가 그래서 코드를 이해하기 어렵다고 생각했던 것 같아요 :)

```suggestion
    await input.userInputsAsync();
    this.#carArray = input.parseCars();
    this.#tryCount = input.parseTryCount();
```

혹은 이런식으로만 수정해도, 
첫 줄은 **`input` 클래스에서 입력값을 (여러개) 받는다**
두번째 줄은 **`input` 클래스에서 자동차 parse**
세번째 줄은 **`input` 클래스에서 시도 횟수 parse**

요런식으로 조금 더 이해가 수월해질 수도 있을 것 같아요. `get`이라는 네이밍 때문에 뭔가 반환값을 받아와야할 것 같은 느낌이 강해지는 것 같아요..!
여러 의견과 방법이 있겠군요! 😄